utils/flink-components-testkit/src/main/scala/pl/touk/nussknacker/engine/flink/util/test/FlinkTestScenarioRunner.scala

```diff
@@ -172,17 +178,34 @@ class FlinkTestScenarioRunner(
 
         env.executeAndWaitForFinished(scenario.name.value)()
 
-        RunUnitResult(errors = testScenarioCollectorHandler.resultsCollectingListener.results.exceptions)
+        TestResultSinkFactory.extractSinkOutputFor(testExtensionsHolder.runId) match {
```

Comment:
Can you write a short explanation what's going on here in the code? It looks hacky. Also I think that it would be easier to understand if you'd show the main path and some fallback because it looks like you trying to solve such a case. Am I right?